src/components/header/Header.component.tsx

```diff
@@ -9,52 +9,62 @@ import LightThemeBtn from '@/assets/sun.svg';
 
 import styles from './header.module.css';
 
-export const HeaderComponent = () => (
-    <header className={styles.header}>
-        <div className={styles.wrapper}>
-            <img className={styles.logo} src={logo} alt="logo" />
-            <div className={styles.header__themes}>
-                <button className={`${styles.header__themeBtn} ${styles.header__themeBtnactive}`}>
-                    <img className={styles.header__themeimg} src={LightThemeBtn} alt="LightTheme" />
-                </button>
-                <img className={styles.divider} src={Divider} alt="Divider" />
-                <button className={styles.header__themeBtn}>
-                    <img className={styles.header__themeimg} src={DarkThemeBtn} alt="DarkTheme" />
-                </button>
-            </div>
+interface HeaderProps {
+    toggleAbout: () => void;
+    toggleProducts: () => void;
+}
+function HeaderComponent({ toggleAbout, toggleProducts }: HeaderProps) {
+    return (
+        <>
+            <header className={styles.header}>
+                <div className={styles.wrapper}>
+                    <img className={styles.logo} src={logo} alt="logo" />
+                    <div className={styles.header__themes}>
+                        <button className={`${styles.header__themeBtn} ${styles.header__themeBtnactive}`}>
+                            <img className={styles.header__themeimg} src={LightThemeBtn} alt="LightTheme" />
+                        </button>
+                        <img className={styles.divider} src={Divider} alt="Divider" />
+                        <button className={styles.header__themeBtn}>
+                            <img className={styles.header__themeimg} src={DarkThemeBtn} alt="DarkTheme" />
+                        </button>
+                    </div>
 
-            <nav className={styles.header__nav}>
-                <ul className={styles.header__navwrapper}>
-                    <li>
-                        <a className={styles.header__linkactive} href="/">
-                            About
-                        </a>
-                    </li>
-                    <li>
-                        <a className={styles.header__link} href="/">
-                            Products
-                        </a>
-                    </li>
-                </ul>
-            </nav>
+                    <nav className={styles.header__nav}>
+                        <ul className={styles.header__navwrapper}>
+                            <li>
+                                <button className={styles.header__linkactive} onClick={toggleAbout}>
+                                    About
+                                </button>
+                            </li>
+                            <li>
+                                <button className={styles.header__link} onClick={toggleProducts}>
```

Comment:
Please add the logic to change the active state of nav menu 
<img width="1398" alt="image" src="https://github.com/demasha0908/fe-react-2024/assets/98031768/108cbba2-17ce-455b-b563-68087ea1bb1e">

Reply:
fixed